test/conformance/blue_green_test.go

```diff
@@ -107,9 +107,22 @@ func TestBlueGreenRoute(t *testing.T) {
 		t.Fatalf("Error fetching Route %s: %v", names.Route, err)
 	}
 
-	// TODO(tcnghia): Add a RouteStatus method to create this string.
-	blueDomain := fmt.Sprintf("%s.%s", blue.TrafficTarget, route.Status.Domain)
-	greenDomain := fmt.Sprintf("%s.%s", green.TrafficTarget, route.Status.Domain)
+	var blueDomain, greenDomain string
+	for _, tt := range route.Status.Traffic {
+		if tt.Name == blue.TrafficTarget {
+			// Strip prefix as WaitForEndPointState expects a domain
+			// without scheme.
+			blueDomain = strings.TrimPrefix(tt.URL, "http://")
```

Comment:
Perhaps use `"net/url"` to parse the URL and extract the host?

Reply:
I prefer this for now as it is the opposite of what "WaitForEndpoingState" is doing. However, "WaitForEndpointState" needs to be fixed.

I would like to follow up and update pkg to not do `url.Parse(fmt.Sprintf("http://%s", theURL))`. There is a fallback to `asURL, err = url.Parse(theURL)`; however, due to the parsing behavior the fallback does not actually work.